lib/ngx/semaphore.lua

```diff
@@ -110,7 +110,10 @@ function _M.wait(self, seconds)
         return nil, "timeout"
     end
 
-    return co_yield()
+    -- prevent the semaphore instance to be garbage collected
+    -- when someone is waiting on it
+    local ok, err = co_yield()
+    return ok, err
```

Comment:
`coroutine.yield()` never returns so we can just omit the `return` part to avoid triggering Lua's tail-call optimization (TCO).